src/compiler/checker.ts

```diff
@@ -6761,17 +6761,20 @@ namespace ts {
             return links.resolvedSignature;
         }
 
+        /**
+         * A JS function gets a synthetic rest parameter if it references `arguments` AND:
+         * 1. It has no parameters and at least one `@param` and the first `@param` has a type and the type starts with `...`

```

Comment:
Comment says `the first @param`, but we are looping over all of them.

Reply:
Good catch. I decided to change the comment in order to continue sharing most of the code here.